lib/orderbook/MatchingEngine.ts

```diff
@@ -152,9 +152,12 @@ class MatchingEngine {
     return matchingResult;
   }
 
-  public removeOwnOrder = (orderId: string): orders.StampedOwnOrder | null => {
-    return this.priorityQueues.buyOrders.removeOne(order => order.id === orderId) ||
-      this.priorityQueues.sellOrders.removeOne(order => order.id === orderId);
+  public removeOwnOrder = (orderId: string): orders.StampedOrder | null => {
```

Comment:
Why `removeOwnOrder` and `removePeerOrder`? We should either unify them or change the return types to `orders.StampedOwnOrder` and `orders.StampedPeerOrder` because they are doing the same thing.

Reply:
I agree that this design is weird. I did it as a preparation to #215, which would require to have them separated. It would require us to redo things anyway.

Reply:
Why not cast the order we return as a `StampedOwnOrder`?